src/libServer/ScillaIPCServer.cpp

```diff
@@ -232,7 +232,7 @@ bool ScillaIPCServer::fetchBlockchainInfo(const std::string &query_name,
   }
 
   TxBlockSharedPtr txBlockSharedPtr;
-  if (query_name == "BLOCKCOINBASE" || query_name == "BLOCKTIMESTAMP" ||
+  if (query_name == "BLOCKHASH" || query_name == "BLOCKCOINBASE" || query_name == "BLOCKTIMESTAMP" ||
```

Comment:
This part of code needs more protective programming (refactor), in case a new or unknown query request enters here it is caught at all.